tools/cache_warm.cc

```diff
@@ -0,0 +1,280 @@
+                        //  (likely the copy to vnode is not needed, but feels better
+                        chr_ptr=dirname(buffer);
+                        if (NULL!=chr_ptr)
+                            strcpy(vnode, chr_ptr);
+                        else
+                            error_seen=true;
+                    }   // if
+
+                    // good line
+                    if (!error_seen && !ignore_file)
+                    {
+                        // did vnode change
+                        if (0!=cur_path.compare(vnode))
+                        {
+                            // write previous
+                            if (!cur_path.empty())
+                            {
+                                CowFile cow(cur_path, record);
+
+                                if (!cow.GetStatus().ok())
+                                {
+                                    fprintf(stderr, "CowFile failed.\n");
+                                    error_seen=true;
+                                }   // if
+                            }   // if
+
+                            record.clear();
+                            cur_path=vnode;
+                        }   // if
+
+                        // add file info to record
+                        if (!error_seen)
+                        {
+                            file_no=atol(file);
+                            leveldb::PutVarint32(&record, leveldb::VersionEdit::kFileCacheObject);
+                            leveldb::PutVarint32(&record, level);
+                            leveldb::PutVarint64(&record, file_no);
+                            leveldb::PutVarint64(&record, file_size);
+                        }   // if
+                    }   // if
+                    else if (!ignore_file)
+                    {
+                        fprintf(stderr, "Unable to parse line in %s at line number %zd\n",
+                                *cursor, line_count);
+                    }   // else
+                }   // while
+
+                // write last
+                if (!error_seen && !cur_path.empty())
+                {
+                    CowFile cow(cur_path, record);
+
+                    if (!cow.GetStatus().ok())
+                    {
+                        fprintf(stderr, "CowFile failed.\n");
+                        error_seen=true;
+                    }   // if
+                }   // if
+
+                fclose(list);
+                list=NULL;
+            }   // if
+            else
+            {
+                error_seen=true;
+                fprintf(stderr, "Unable to open file %s (error %d)",
+                        *cursor, errno);
+            }   // else
+        }   // else
+    }   // for
+
+    // cleanup
+    leveldb::Env::Default()->Shutdown();
+//    delete options.filter_policy;
+
+    if (1==argc)
+        command_help();
+
+    return( error_seen ? 1 : 0 );
+
+}   // main
+
+
+void
+command_help()
```

Comment:
Paul Place writes:  need to update the comman_help() method from sst_rewrite to cache_warm